File: tests/fixtures/languages/ruby/users_controller.rb
class User < ApplicationRecord
  validates :name, presence: true
  validates :email, presence: true, uniqueness: true
end

class UsersController < ApplicationController
  def index
    @users = User.all
    render json: { items: @users, count: @users.count }
  end

  def show
    @user = User.find(params[:id])
    render json: { item: @user }
  rescue ActiveRecord::RecordNotFound
    render json: { error: "Not found" }, status: :not_found
  end

  def create
    @user = User.new(user_params)
    if @user.save
      render json: { item: @user }, status: :created
    else
      render json: { errors: @user.errors }, status: :unprocessable_entity
    end
  end

  def update
    @user = User.find(params[:id])
    if @user.update(user_params)
      render json: { item: @user }
    else
      render json: { errors: @user.errors }, status: :unprocessable_entity
    end
  rescue ActiveRecord::RecordNotFound
    render json: { error: "Not found" }, status: :not_found
  end

  def destroy
    @user = User.find(params[:id])
    @user.destroy
    render json: { message: "Deleted" }
  rescue ActiveRecord::RecordNotFound
    render json: { error: "Not found" }, status: :not_found
  end

  def search
    query = params[:q] || ""
    @users = User.where("name LIKE ?", "%#{query}%")
    render json: { results: @users }
  end

  private

  def user_params
    params.require(:user).permit(:name, :email)
  end

  def unused_helper
    "dead code"
  end
end

File: tests/fixtures/languages/csharp/Controllers/UsersController.cs
using Microsoft.AspNetCore.Mvc;
using System.Collections.Generic;
using System.Linq;

namespace Demo.Controllers;

[ApiController]
[Route("api/[controller]")]
public class UsersController : ControllerBase
{
    private static readonly List<User> _users = new();
    private static int _nextId = 1;

    [HttpGet]
    public ActionResult<IEnumerable<User>> GetAll()
    {
        return Ok(new { items = _users, count = _users.Count });
    }

    [HttpGet("{id}")]
    public ActionResult<User> GetById(int id)
    {
        var user = _users.FirstOrDefault(u => u.Id == id);
        if (user == null) return NotFound(new { error = "Not found" });
        return Ok(new { item = user });
    }

    [HttpPost]
    public ActionResult<User> Create(User user)
    {
        user.Id = _nextId++;
        _users.Add(user);
        return CreatedAtAction(nameof(GetById), new { id = user.Id }, new { item = user });
    }

    [HttpPut("{id}")]
    public ActionResult<User> Update(int id, User updated)
    {
        var user = _users.FirstOrDefault(u => u.Id == id);
        if (user == null) return NotFound(new { error = "Not found" });
        user.Name = updated.Name;
        user.Email = updated.Email;
        return Ok(new { item = user });
    }

    [HttpDelete("{id}")]
    public ActionResult Delete(int id)
    {
        var user = _users.FirstOrDefault(u => u.Id == id);
        if (user == null) return NotFound(new { error = "Not found" });
        _users.Remove(user);
        return Ok(new { message = "Deleted" });
    }

    [HttpGet("search")]
    public ActionResult Search([FromQuery] string q)
    {
        var results = _users.Where(u => u.Name.Contains(q)).ToList();
        return Ok(new { results });
    }

    private void UnusedHelper()
    {
    }
}

public class User
{
    public int Id { get; set; }
    public string Name { get; set; } = "";
    public string Email { get; set; } = "";
}
